src/main.c

```diff
@@ -124,6 +113,44 @@ int run_ntttcp_sender(struct ntttcp_test_endpoint *tep)
 	ASPRINTF(&log, "%d threads created", threads_created);
 	PRINT_INFO_FREE(log);
 
+	// wait for all connections created
+	conns_total = test->server_ports * test->threads_per_server_port * test->conns_per_thread;
+	while (conns_creation_time < SEC_TO_USEC) {
+		conns_created = 0;
+		gettimeofday(&now, NULL);
+		conns_creation_time = (now.tv_sec - start_time.tv_sec) * SEC_TO_USEC + now.tv_usec - start_time.tv_usec;
+		for (t = 0; t < test->server_ports; t++) {
+			for (n = 0; n < test->threads_per_server_port; n++ ) {
+				cs = tep->client_streams[t * test->threads_per_server_port + n];
+				conns_created += cs->num_conns_created;
+			}
+		}
+		if (conns_created == conns_total) {
+			ASPRINTF(&log, "%d connections created in %ld microseconds", conns_created, conns_creation_time);
+			PRINT_INFO_FREE(log);
+			break;
+		}
+	}
+	if (conns_created != conns_total) {
+		ASPRINTF(&log, "%d connections created in %ld microseconds", conns_created, conns_creation_time);
+		PRINT_ERR_FREE(log);
+	}
+
+	// request receiver to start the test
```

Comment:
do this only in the case of sync. so below code should be wrapped with "if (test->no_synch == false ) {"